### PR TITLE
Secondary listening on TCP port

### PR DESCRIPTION
**Pull request by lbonn**
Based on #622.

The actual messages and parsing need to be done, for now please treat the content of `aktualizr_secondary_ipc.h` as a placeholder.

Not sure how to test this.

### CODE REVIEW

File: src/aktualizr_secondary/main.cc [hunk 1]
```diff
@@ -72,6 +84,59 @@ bpo::variables_map parse_options(int argc, char *argv[]) {
...
+  if (setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &reuseaddr, sizeof(reuseaddr)) < 0) {
+    throw std::runtime_error("setsockopt(SO_REUSEADDR) failed");
+  }
+
+  if (bind(socket_fd, reinterpret_cast<const sockaddr *>(&sa), sizeof(sa)) < 0) {
+    throw std::runtime_error("bind failed");
+  }
+
+  if (listen(socket_fd, SOMAXCONN) < 0) {
+    throw std::runtime_error("listen failed");
```
Comment (patriotyk) [inline src/aktualizr_secondary/main.cc]: Maybe `close(socket_fd)` on error?

**Comment by lbonn** [comment]
coverage went crazy again, I'll look into that

**Comment by cajun-rat** [comment]
(rebased)

**Comment by cajun-rat** [comment]
Approved, assuming CI passes

**Comment by lbonn** [comment]
I pushed another commit on top of that, sorry

**Comment by cajun-rat** [comment]
Heh, never mind :)

**Pull request merged**